example_test.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package credentials_test
+
+import (
+	"context"
+	"fmt"
+
+	credentials "github.com/oras-project/oras-credentials-go"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+func ExampleNewNativeStore() {
+	// The argument of NewNativeStore can be the native keychains
+	// ("wincred" for Windows, "pass" for linux and "osxkeychain" for macOS),
+	// or any program that follows the docker-credentials-helper protocol.
+	// See: https://docs.docker.com/engine/reference/commandline/login#credentials-store
+	ns := credentials.NewNativeStore("pass")
+
+	// save credentials into the store
+	err := ns.Put(context.Background(), "localhost:8080",
+		auth.Credential{
+			Username: "username-example",
+			Password: "password-example"})
+	if err != nil {
+		panic(err)
+	}
+
+	// get credentials from the store
+	cred, err := ns.Get(context.Background(), "localhost:8080")
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println(cred)
+
+	// delete the credentials from the store
+	err = ns.Delete(context.Background(), "localhost:8080")
+	if err != nil {
+		panic(err)
+	}
+}
+
+func ExampleNewFileStore() {
+	fs, err := credentials.NewFileStore("example/path/config.json")
+	if err != nil {
+		panic(err)
+	}
+
+	// save credentials into the store
+	err = fs.Put(context.Background(), "localhost:8080",
+		auth.Credential{
+			Username: "username-example",
+			Password: "password-example"})
+	if err != nil {
+		panic(err)
+	}
+
+	// get credentials from the store
+	cred, err := fs.Get(context.Background(), "localhost:8080")
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println(cred)
+
+	// delete the credentials from the store
+	err = fs.Delete(context.Background(), "localhost:8080")
+	if err != nil {
+		panic(err)
+	}
+}
+
+func ExampleNewStore() {
+	// NewStore returns a Store based on the given configuration file. It will
+	// automatically determine which Store (file store or native store) to use.
+	// If the native store is not available, you can save your credentials in
+	// the configuration file by specifying AllowPlaintextPut: true, but keep
+	// in mind that this is an unsafe workaround.
+	// See the documentation for details.
+	s, err := credentials.NewStore("example/path/config.json", credentials.StoreOptions{AllowPlaintextPut: true})
```

Comment:
nit: `s` looks a bit too short for an example. Might be better to call it `store`.

Reply:
changed.